hs_file_types/models/raster.py

```diff
@@ -0,0 +1,528 @@
+        """
+            Sets a tif or zip raster resource file to GeoRasterFile type
+            :param resource: an instance of resource type CompositeResource
+            :param file_id: id of the resource file to be set as GeoRasterFile type
+            :param user: user who is setting the file type
+            :return:
+            """
+
+        # had to import it here to avoid import loop
+        from hs_core.views.utils import create_folder, move_or_rename_file_or_folder
+
+        log = logging.getLogger()
+
+        # get the file from irods
+        res_file = utils.get_resource_file_by_id(resource, file_id)
+
+        # base file name (no path included)
+        file_name = utils.get_resource_file_name_and_extension(res_file)[1]
+        # file name without the extension
+        file_name = file_name.split(".")[0]
+
+        metadata = []
+        if res_file is not None and res_file.has_generic_logical_file:
+            # get the file from irods to temp dir
+            temp_file = utils.get_file_from_irods(res_file)
+            # validate the file
+            error_info, files_to_add_to_resource = _raster_file_validation(raster_file=temp_file)
+            if not error_info:
+                log.info("Geo raster file type file validation successful.")
+                # extract metadata
+                temp_dir = os.path.dirname(temp_file)
+                temp_vrt_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if
+                                      '.vrt' == os.path.splitext(f)[1]].pop()
+                res_md_dict = raster_meta_extract.get_raster_meta_dict(temp_vrt_file_path)
+                wgs_cov_info = res_md_dict['spatial_coverage_info']['wgs84_coverage_info']
+                # add core metadata coverage - box
+                if wgs_cov_info:
+                    box = {'coverage': {'type': 'box', 'value': wgs_cov_info}}
+                    metadata.append(box)
+
+                # Save extended meta spatial reference
+                ori_cov = {'OriginalCoverage': {
+                    'value': res_md_dict['spatial_coverage_info']['original_coverage_info']}}
+                metadata.append(ori_cov)
+
+                # Save extended meta cell info
+                res_md_dict['cell_info']['name'] = os.path.basename(temp_vrt_file_path)
+                metadata.append({'CellInformation': res_md_dict['cell_info']})
+
+                # Save extended meta band info
+                for band_info in res_md_dict['band_info'].values():
+                    metadata.append({'BandInformation': band_info})
+
+                log.info("Geo raster file type metadata extraction was successful.")
+                with transaction.atomic():
+                    # first delete the raster file that we retrieved from irods
+                    # for setting it to raster file type
+                    delete_resource_file(resource.short_id, res_file.id, user)
```

Comment:
@pkdash I understand why original resource files needs to be deleted now. However, since no metadata had been extracted from the original file at this point, delete_resource_file_only() call may be better applied here which does not send signals.